### PR TITLE
Make reload non-blocking in feedlist

### PR DESCRIPTION
**Pull request by dennisschagt**
Partial fix for:
- https://github.com/newsboat/newsboat/issues/75
- https://github.com/akrennmair/newsbeuter/issues/606

It is probably possible to apply the same fix to make reload non-blocking in the ItemListFormAction.
However, I still need to verify this does not have unintended consequences.

The current changes have the effect that single-feed reloads will also start sending notifications (related to `notify-program`, before, notifications were only sent when finishing a `reload-all` action). To reproduce, add the following to the config file:
```
notify-program "notify-send"
notify-always yes
```
I think that should not happen, do others agree?

Marking as draft as I need to verify there are no other unintended changes.

### CODE REVIEW

**Comment by Minoru** [comment]
> I think that should not happen, do others agree?

I agree. Reloading a single feed is expected to finish quickly, so a notification will be redundant most of the time.

File: src/itemlistformaction.cpp [hunk 1]
```diff
@@ -408,7 +408,7 @@ bool ItemListFormAction::process_operation(Operation op,
 		if (!show_searchresult) {
 			LOG(Level::INFO,
 				"ItemListFormAction: reloading current feed");
-			v->get_ctrl()->get_reloader()->reload(pos);
+			v->get_ctrl()->get_reloader()->start_reload_thread(pos);
 			invalidate_everything();
```
Comment (dennisschagt) [inline src/itemlistformaction.cpp:412]: We will probably have to call `invalidate_everything()` and force an STFL redraw when the reload finishes.
I'm not yet sure how to properly do that:
- not sure if STFL can be safely instructed to redraw from a background thread
- the ItemListFormAction might have been closed during the reload

**Comment by dennisschagt** [comment]
I'm currently not interested in spending a lot of time on this feature.
This might result in this PR being open (as a draft) for a long time.

@Minoru: If you ever want to clean up open PRs, feel free to close this one.
I can always reopen it when I have a full solution.

File: src/itemlistformaction.cpp (hunk 1 above)
Comment (Minoru) [reply src/itemlistformaction.cpp:412]: FWIW, `Reloader` already updates current formaction's `msg`, ignoring the first point and circumventing the second one.

**Comment by Minoru** [comment]
Okay. Thanks for the research you've done already!

**Comment by dennisschagt** [comment]
Closing this PR as I don't plan to work on this feature anytime soon.

**Pull request closed without merging**